src/asm/parser.y

```diff
@@ -1169,6 +1171,40 @@ ds		: T_POP_DS uconst { sect_Skip($2, true); }
 			sect_RelBytes($2, $4.args, $4.nbArgs);
 			freeDsArgList(&$4);
 		}
+		| T_POP_DS ds_align trailing_comma {
+			sect_Skip($2.nbBytes, true);
+			sect_AlignPC($2.alignment, $2.alignOfs);
+		}
+		| T_POP_DS ds_align T_COMMA ds_args trailing_comma {
+			sect_RelBytes($2.nbBytes, $4.args, $4.nbArgs);
+			sect_AlignPC($2.alignment, $2.alignOfs);
+			freeDsArgList(&$4);
+		}
+;
+
+ds_align	: T_OP_ALIGN T_LBRACK uconst T_RBRACK {
+			if ($3 > 16) {
+				error("Alignment must be between 0 and 16, not %u\n", $3);
+			} else {
+				uint32_t mask = (1 << $3) - 1;
+
+				// TODO: correct formula; sym_GetPCValue() won't work in floating sections
+				$$.alignment = $3;
+				$$.nbBytes = (mask + 1 - (sym_GetPCValue() & mask)) & mask;
```

Comment:
I'd pass the arguments as-is, and perform the byte count calc in a factored-out function, instead.